source/common/router/router.cc

```diff
@@ -1045,9 +1045,23 @@ void Filter::onStreamMaxDurationReached(UpstreamRequest& upstream_request) {
 
   callbacks_->streamInfo().setResponseFlag(
       StreamInfo::ResponseFlag::UpstreamMaxStreamDurationReached);
+  // Grab the const ref to call the const method of StreamInfo.
+  const auto& stream_info = callbacks_->streamInfo();
+  auto maybe_request_timeout_code =
+      stream_info.downstreamTiming().has_value() &&
+              stream_info.downstreamTiming()
+                  .value()
+                  .get()
+                  .lastDownstreamRxByteReceived()
+                  .has_value() &&
+              Runtime::runtimeFeatureEnabled(
+                  "envoy.reloadable_features.override_request_timeout_by_gateway_timeout")
+          ? Http::Code::GatewayTimeout
+          : Http::Code::RequestTimeout;
```

Comment:
Can you put this logic into a shared utility function somewhere? It's duplicated in the other file.

Reply:
Ping on this?

/wait

Reply:
Working on it. It's weird that this comment shows up and disappears on UI.

Reply:
resolved in https://github.com/envoyproxy/envoy/pull/20026/commits/64f0076e3685daf4405f0119e28f99de2c6af9e8